src/patternfly/components/Table/table.scss

```diff
@@ -954,39 +924,35 @@
 
 // Check table cell
 .#{$table}__check {
-  --#{$table}--cell--FontSize: var(--#{$table}__check--input--FontSize);
+  --#{$table}--cell--PaddingLeft: var(--#{$table}__check--PaddingLeft);
+  --#{$table}--cell--PaddingRight: var(--#{$table}__check--PaddingRight);
 
-  label:not([disabled]),
-  input:not([disabled]) {
-    cursor: pointer;
-  }
-
-  label {
+  .#{$check}.pf-m-standalone {
     display: inline-flex;
-    padding: inherit;
-    margin-block-start: calc(var(--#{$table}--cell--PaddingTop) * -1);
-    margin-block-end: calc(var(--#{$table}--cell--PaddingBottom) * -1);
-    margin-inline-start: calc(var(--#{$table}--cell--PaddingLeft) * -1);
-    margin-inline-end: calc(var(--#{$table}--cell--PaddingRight) * -1);
 
+    thead & {
+      vertical-align: bottom;
+    }
+
+    tbody & {
+      vertical-align: top;
+    }
```

Comment:
just a thought, but you could probably add `vertical-align: top` as the base style [under the display declaration](https://github.com/patternfly/patternfly/blob/a01f949cbef42c7bc6a7cf7e9b3fc64099189fc5/src/patternfly/components/Table/table.scss#L931) and just change that in `thead`?